tests/unit/recommenders/evaluation/test_python_evaluation.py

```diff
@@ -100,12 +67,31 @@ def rating_pred_binary(rating_pred):
 
 
 def test_column_dtypes_match(rating_true, rating_pred):
-    # Change data types of true and prediction data, and there should type error produced
-    rating_true[DEFAULT_USER_COL] = rating_true[DEFAULT_USER_COL].astype(str)
-    rating_true[DEFAULT_RATING_COL] = rating_true[DEFAULT_RATING_COL].astype(str)
+    mocked_fn = Mock(return_value=True)
```

Comment:
A question here, what is the function of the mock here?

Reply:
that is to test if `_check_column_dtypes` returns the same output as the input function when there's no exceptions.
I'll add a comment to the test.